cupyx/scipy/interpolate/_bspline2.py

```diff
@@ -554,7 +554,8 @@ def _make_periodic_spline(x, y, t, k, axis):
     extradim = prod(y.shape[1:])
     rhs = cupy.empty((n + k - 1, extradim), dtype=float)
     rhs[:(k - 1), :] = 0
-    rhs[(k - 1):, :] = y.reshape((-1, extradim))
+    rhs[(k - 1):, :] = (y.reshape(n, 0) if y.size == 0 else
+                        y.reshape((-1, extradim)))
```

Comment:
@ev-br, I had to do this modification in order to fix the `test_zero_size`, which otherwise would fail with the following stacktrace:

```python
Traceback (most recent call last):
  File "/home/andfoy/Documentos/Quansight/cupy/cupy/testing/_loops.py", line 45, in _call_func
    result = impl(*args, **kw)
  File "/home/andfoy/Documentos/Quansight/cupy/tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py", line 389, in test_zero_size
    obj = cls(x, y, bc_type=bc_type)
  File "/home/andfoy/Documentos/Quansight/cupy/cupyx/scipy/interpolate/_bspline2.py", line 251, in make_interp_spline
    return _make_periodic_spline(x, y, t, k, axis)
  File "/home/andfoy/Documentos/Quansight/cupy/cupyx/scipy/interpolate/_bspline2.py", line 557, in _make_periodic_spline
    rhs[(k - 1):, :] = y.reshape((-1, extradim))
  File "cupy/_core/core.pyx", line 765, in cupy._core.core._ndarray_base.reshape
    return _manipulation._ndarray_reshape(self, shape, order)
  File "cupy/_core/_routines_manipulation.pyx", line 83, in cupy._core._routines_manipulation._ndarray_reshape
    return _reshape(self, shape)
  File "cupy/_core/_routines_manipulation.pyx", line 352, in cupy._core._routines_manipulation._reshape
    shape = internal.infer_unknown_dimension(shape_spec, self.size)
  File "cupy/_core/internal.pyx", line 208, in cupy._core.internal.infer_unknown_dimension
    raise ValueError(
ValueError: cannot reshape array of size 0 into shape (-1, 0)
```